src/ragger/navigator/interface.py

```diff
@@ -0,0 +1,297 @@
+"""
+   Copyright 2022 Ledger SAS
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+from abc import ABC
+from enum import Enum, auto
+from pathlib import Path
+from typing import List, Optional, Dict, Any
+from time import sleep, time
+
+from ragger.backend import BackendInterface, SpeculosBackend
+from ragger.firmware import Firmware
+from ragger.utils import Crop
+
+
+class NavInsID(Enum):
+    WAIT = auto()
+
+    # Navigation instructions for Nano devices
+    RIGHT_CLICK = auto()
+    LEFT_CLICK = auto()
+    BOTH_CLICK = auto()
+
+
+class NavIns:
+
+    def __init__(self, id: NavInsID, args=(), kwargs: Dict[str, Any] = {}):
+        self.id = id
+        self.args = args
+        self.kwargs = kwargs
+
+
+class NavigatorInterface(ABC):
```

Comment:
@xchapron-ledger I think it shouldn't be called "Interface" as this class implements stuff. It's just a parent class to a device navigator... So you might just call it "Navigator" and rename the file `interface.py` to `navigator.py`

Reply:
done